### DISCUSSION TITLE
Questions about private keys

### DISCUSSION THREAD

**Discussion by JarrodWoodard** [ecadlabs/signatory#159] [Q&A]
Can keys generated on a YubiHSM 2, be securely backed up in a written format, like a seed phrase? ( in case of a disaster, loss or theft )

Can you import a seed phrase from a Ledger Nano S into a YubiHSM 2?

How is the YubiHSM 2, secure from physical access, is it password or PIN protected somehow? 

If so, since it does not have a direct interface like a Ledger Nano S, is that password/PIN vulnerable to a compromised system? 

Thank you so much whoever takes the time to respond!

**Comment by jevonearth** [comment]
Hi @JarrodWoodard, Thank you for your questions.

> Can keys generated on a YubiHSM 2, be securely backed up in a written format, like a seed phrase? ( in case of a disaster, loss or theft )

- You can import a secret key into the YubiHSM device, but, to my knowledge, it does not support importing of BIP39 mnemonics (seed phrases). You should be able to do this with third party tools.
- You can not generate a secret on the yubiHSM and then extract it to plain text.
- There is a backup/restore procedure for the YubiHSM which you can find here: https://developers.yubico.com/YubiHSM2/Usage_Guides/YubiHSM_2_Windows_Deployment_Guide--Configure_YubiHSM_2_Key_Storage_Provider_for_Microsoft_Windows_Server/Back_Up_and_Restore_Key_Material.html

> Can you import a seed phrase from a Ledger Nano S into a YubiHSM 2?

- Not with out third party tools. You can import a secret key, but not a BIP39 mnemonic phrase. You will need to convert your seed phrase to a secret key with other tools.
- A common strategy is to generate a new secret on an airgapped system. Record the secret to a off-line medium such as paper. Import the secret key into the YubiHSM _on the airgapped system_. Move the YubiHSM2 device to your baker system. Securely wipe the airgapped system.

> How is the YubiHSM 2, secure from physical access, is it password or PIN protected somehow?

- Physical security of the yubiHSM device is the operators responsibility
- The YubiHSM does have an access password. This password should be reset from the default value. See the YubiHSM Admin Guide for instructions on updating the access password.
- Signatory requires this password to access the YubiHSM. Using the password feature is recommended, but it does not remove the requirement for robust physical and digital security.
- YubiHSM's do not reset after repeated failed password attempts in the same way that Ledger devices do.

> If so, since it does not have a direct interface like a Ledger Nano S, is that password/PIN vulnerable to a compromised system?

See above

**Reply by JarrodWoodard** [reply]
Thank you very much! 

> YubiHSM's do not reset after repeated failed password attempts in the same way that Ledger devices do.

That is pretty concerning and different from my understanding of how a regular Yubikey works. I hope there is some way to setup this feature, I'll inquire with Yubico.

**Reply by JarrodWoodard** [reply]
It looks as though the Yubico HSM 2 is admin accessed via a regular a Yubikey. Seemingly to grant the best of both worlds. The function of the HSM cannot be attacked via intentionally putting in the wrong password / pin while the Yubikey allowing access to the HSM will lock itself out if the wrong pin is entered too many times.
